benchmarks/CMakeLists.txt

```diff
@@ -2,8 +2,32 @@ find_package(Boost 1.56.0 REQUIRED COMPONENTS program_options)
 
 # FIXME: when we start requiring benchmark >= 1.8.0, reenable
 # "readability-redundant-string-cstr" check in .clang-tidy
-find_package(benchmark 1.5.4 REQUIRED)
-message(STATUS "Found benchmark: ${benchmark_DIR} (version \"${benchmark_VERSION}\")")
+find_package(benchmark QUIET 1.5.4)
+if(benchmark_FOUND)
+  message(STATUS "Found benchmark: ${benchmark_DIR} (version \"${benchmark_VERSION}\")")
+else()
+  message(STATUS "benchmark not found, fetching from GitHub")
+  include(FetchContent)
+  set(BENCHMARK_ENABLE_TESTING OFF)
+
+  list(APPEND CMAKE_MESSAGE_INDENT "[benchmark] ")
+  FetchContent_Declare(
+    googlebenchmark
+    DOWNLOAD_EXTRACT_TIMESTAMP FALSE
+    URL https://github.com/google/benchmark/archive/refs/tags/v1.8.0.tar.gz
```

Comment:
Can you comment on the version you picked?

Reply:
I decided to up the version so that when we fix the comment about (about reenabling clang-tidy check), we would not need to change this. No other reason. The latest Google Benchmark is 1.8.3, so same major.